syntax/vm/vm_stdlib_test.go

```diff
@@ -39,6 +39,51 @@ func TestVM_Stdlib(t *testing.T) {
 		{"encoding.from_yaml nil field", "encoding.from_yaml(`foo: null`)", map[string]interface{}{"foo": nil}},
 		{"encoding.from_yaml nil array element", `encoding.from_yaml("[0, null]")`, []interface{}{0, nil}},
 		{"encoding.from_base64", `encoding.from_base64("Zm9vYmFyMTIzIT8kKiYoKSctPUB+")`, string(`foobar123!?$*&()'-=@~`)},
+
+		// Map tests
+		{
+			// Basic case. No conflicting key/val pairs.
+			"targets.merge",
+			`targets.merge([{"a" = "a1", "b" = "b1"}], [{"a" = "a1", "c" = "c1"}], ["a"])`,
```

Comment:
What is the result for invalid input? Imagine the first is a valid map list but the second is nil/invalid type/etc? I imagine it would return the map list, whereas if it is the reverse it would return a blank list?

Reply:
There is an error if the input is in valid:

<img width="1133" alt="Screenshot 2024-10-31 at 18 16 42" src="https://github.com/user-attachments/assets/32dadf5c-9700-4338-9a31-e08d32ca47f1">
<img width="1511" alt="Screenshot 2024-10-31 at 18 16 09" src="https://github.com/user-attachments/assets/0c443fe3-bf54-41bc-97ce-728065e14b72">
<img width="438" alt="Screenshot 2024-10-31 at 18 17 41" src="https://github.com/user-attachments/assets/bd68cd59-6663-4896-b3ca-0e2f6893602c">